### DISCUSSION TITLE
does twind support static site generation

### DISCUSSION THREAD

**Discussion by aakashvaa** [tw-in-js/twind#514] [Q&A]
i'm curious weather twind support static site generation or not. It will be amazing if it does all the processing at build time to pre-generate the css.
pigment css(a css-in-js) has implemented this feature.
https://mui.com/blog/introducing-pigment-css/

**Comment by rschristian** [comment]
You absolutely can, yes. Something like this would suffice:

```js
import { install, extract } from '@twind/core';

const tw = install(userConfig, true);
const { html, css } = extract(`<div id="app" class="color-blue-500">...</div>`, tw);
```

**Reply by aakashvaa** [reply]
thank you

**Reply by aakashvaa** [reply]
is there any blog where i can learn how to properly setup twind in next js.

**Reply by rschristian** [reply]
Not sure, sorry -- I don't use Next.

**Reply by aakashvaa** [reply]
i found that btw thank you for your quick response
